 # Data Collection & Inspection



## Objectives

- Load the raw airline passenger satisfaction dataset.
- Inspect the structure and contents of the dataset.
- Identify any initial data quality issues that may need to be addressed during data cleaning.

## Inputs

- `data/raw_data/train.csv`

## Outputs

- Initial understanding of the airline passenger dataset.
- Identification of potential data cleaning requirements.

---

# Change working directory

In [1]:
import os
current_dir = os.getcwd()
current_dir

'c:\\Users\\hanee\\OneDrive\\Desktop\\Project2\\airline_passenger_satisfaction_analyst\\airline-passenger-satisfaction\\jupyter_notebooks'

In [2]:
if os.path.basename(current_dir) == "jupyter_notebooks":
    os.chdir(os.path.dirname(current_dir))

print(f"Current working directory: {os.getcwd()}")

Current working directory: c:\Users\hanee\OneDrive\Desktop\Project2\airline_passenger_satisfaction_analyst\airline-passenger-satisfaction


Confirm the new current directory

In [3]:
current_dir = os.getcwd()
current_dir

'c:\\Users\\hanee\\OneDrive\\Desktop\\Project2\\airline_passenger_satisfaction_analyst\\airline-passenger-satisfaction'

## Load the dataset

Import Pandas to load and inspect the raw airline passenger dataset.

In [4]:
import pandas as pd

# Initial Data Inspection

Load the raw dataset and inspect the first few rows to understand its structure.

In [41]:
df = pd.read_csv("data/raw_data/train.csv")


### Dataset Inspection

In [42]:
df.head()

,Unnamed: 0,id,Gender,Customer Type,Age,Type of Travel,Class,Flight Distance,Inflight wifi service,Departure/Arrival time convenient,...,Inflight entertainment,On-board service,Leg room service,Baggage handling,Checkin service,Inflight service,Cleanliness,Departure Delay in Minutes,Arrival Delay in Minutes,satisfaction
0,0,70172,Male,Loyal Customer,13,Personal Travel,Eco Plus,460,3,4,...,5,4,3,4,4,5,5,25,18.0,neutral or dissatisfied
1,1,5047,Male,disloyal Customer,25,Business travel,Business,235,3,2,...,1,1,5,3,1,4,1,1,6.0,neutral or dissatisfied
2,2,110028,Female,Loyal Customer,26,Business travel,Business,1142,2,2,...,5,4,3,4,4,4,5,0,0.0,satisfied
3,3,24026,Female,Loyal Customer,25,Business travel,Business,562,2,5,...,2,2,5,3,1,4,2,11,9.0,neutral or dissatisfied
4,4,119299,Male,Loyal Customer,61,Business travel,Business,214,3,3,...,3,3,4,4,3,3,3,0,0.0,satisfied


### Dataset Shape

Check the number of rows and columns in the raw dataset.

In [43]:
df.shape

(103904, 25)

### Column Names

Display all column names to understand the variables available in the dataset.

In [44]:
df.columns.tolist()

['Unnamed: 0',
 'id',
 'Gender',
 'Customer Type',
 'Age',
 'Type of Travel',
 'Class',
 'Flight Distance',
 'Inflight wifi service',
 'Departure/Arrival time convenient',
 'Ease of Online booking',
 'Gate location',
 'Food and drink',
 'Online boarding',
 'Seat comfort',
 'Inflight entertainment',
 'On-board service',
 'Leg room service',
 'Baggage handling',
 'Checkin service',
 'Inflight service',
 'Cleanliness',
 'Departure Delay in Minutes',
 'Arrival Delay in Minutes',
 'satisfaction']

### Data Types

Check the data type of each column to understand how the variables are stored.

In [45]:
df.dtypes

Unnamed: 0                             int64
id                                     int64
Gender                                   str
Customer Type                            str
Age                                    int64
Type of Travel                           str
Class                                    str
Flight Distance                        int64
Inflight wifi service                  int64
Departure/Arrival time convenient      int64
Ease of Online booking                 int64
Gate location                          int64
Food and drink                         int64
Online boarding                        int64
Seat comfort                           int64
Inflight entertainment                 int64
On-board service                       int64
Leg room service                       int64
Baggage handling                       int64
Checkin service                        int64
Inflight service                       int64
Cleanliness                            int64
Departure 

---

### Missing Values

Check each column for missing values before deciding whether any data cleaning is required.

In [46]:
df.isnull().sum()

Unnamed: 0                             0
id                                     0
Gender                                 0
Customer Type                          0
Age                                    0
Type of Travel                         0
Class                                  0
Flight Distance                        0
Inflight wifi service                  0
Departure/Arrival time convenient      0
Ease of Online booking                 0
Gate location                          0
Food and drink                         0
Online boarding                        0
Seat comfort                           0
Inflight entertainment                 0
On-board service                       0
Leg room service                       0
Baggage handling                       0
Checkin service                        0
Inflight service                       0
Cleanliness                            0
Departure Delay in Minutes             0
Arrival Delay in Minutes             310
satisfaction    

### Duplicate Records

Check whether the raw dataset contains any completely duplicated rows.

In [47]:
df.duplicated().sum()

np.int64(0)

## Target Variable

Inspect the values in the `satisfaction` column to understand the prediction target.

In [48]:
df["satisfaction"].value_counts()

satisfaction
neutral or dissatisfied    58879
satisfied                  45025
Name: count, dtype: int64

### Target Distribution

Calculate the percentage of passengers in each satisfaction category.

In [49]:
df["satisfaction"].value_counts(normalize=True) * 100

satisfaction
neutral or dissatisfied    56.666731
satisfied                  43.333269
Name: proportion, dtype: float64

### Dataset Information

Display a summary of the dataset, including column names, non-null values and data types.

In [50]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 103904 entries, 0 to 103903
Data columns (total 25 columns):
 #   Column                             Non-Null Count   Dtype  
---  ------                             --------------   -----  
 0   Unnamed: 0                         103904 non-null  int64  
 1   id                                 103904 non-null  int64  
 2   Gender                             103904 non-null  str    
 3   Customer Type                      103904 non-null  str    
 4   Age                                103904 non-null  int64  
 5   Type of Travel                     103904 non-null  str    
 6   Class                              103904 non-null  str    
 7   Flight Distance                    103904 non-null  int64  
 8   Inflight wifi service              103904 non-null  int64  
 9   Departure/Arrival time convenient  103904 non-null  int64  
 10  Ease of Online booking             103904 non-null  int64  
 11  Gate location                      103904 non-null

---

### Descriptive Statistics

Review summary statistics for the numerical variables to identify their ranges and any values that may require further investigation.

In [51]:
df.describe()

,Unnamed: 0,id,Age,Flight Distance,Inflight wifi service,Departure/Arrival time convenient,Ease of Online booking,Gate location,Food and drink,Online boarding,Seat comfort,Inflight entertainment,On-board service,Leg room service,Baggage handling,Checkin service,Inflight service,Cleanliness,Departure Delay in Minutes,Arrival Delay in Minutes
count,103904.000000,103904.000000,103904.000000,103904.000000,103904.000000,103904.000000,103904.000000,103904.000000,103904.000000,103904.000000,103904.000000,103904.000000,103904.000000,103904.000000,103904.000000,103904.000000,103904.000000,103904.000000,103904.000000,103594.000000
mean,51951.500000,64924.210502,39.379706,1189.448375,2.729683,3.060296,2.756901,2.976883,3.202129,3.250375,3.439396,3.358158,3.382363,3.351055,3.631833,3.304290,3.640428,3.286351,14.815618,15.178678
std,29994.645522,37463.812252,15.114964,997.147281,1.327829,1.525075,1.398929,1.277621,1.329533,1.349509,1.319088,1.332991,1.288354,1.315605,1.180903,1.265396,1.175663,1.312273,38.230901,38.698682
min,0.000000,1.000000,7.000000,31.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,1.000000,0.000000,0.000000,0.000000,0.000000,0.000000
25%,25975.750000,32533.750000,27.000000,414.000000,2.000000,2.000000,2.000000,2.000000,2.000000,2.000000,2.000000,2.000000,2.000000,2.000000,3.000000,3.000000,3.000000,2.000000,0.000000,0.000000
50%,51951.500000,64856.500000,40.000000,843.000000,3.000000,3.000000,3.000000,3.000000,3.000000,3.000000,4.000000,4.000000,4.000000,4.000000,4.000000,3.000000,4.000000,3.000000,0.000000,0.000000
75%,77927.250000,97368.250000,51.000000,1743.000000,4.000000,4.000000,4.000000,4.000000,4.000000,4.000000,5.000000,4.000000,4.000000,4.000000,5.000000,4.000000,5.000000,4.000000,12.000000,13.000000
max,103903.000000,129880.000000,85.000000,4983.000000,5.000000,5.000000,5.000000,5.000000,5.000000,5.000000,5.000000,5.000000,5.000000,5.000000,5.000000,5.000000,5.000000,5.000000,1592.000000,1584.000000


### Delay Statistics

Review the departure and arrival delay variables separately to understand their ranges.

In [52]:
df[["Departure Delay in Minutes", "Arrival Delay in Minutes"]].describe()

,Departure Delay in Minutes,Arrival Delay in Minutes
count,103904.000000,103594.000000
mean,14.815618,15.178678
std,38.230901,38.698682
min,0.000000,0.000000
25%,0.000000,0.000000
50%,0.000000,0.000000
75%,12.000000,13.000000
max,1592.000000,1584.000000


### Observation

Most passengers have little or no delay, as the median delay is 0 minutes for both departure and arrival.

However, the maximum delays are much higher at 1,592 minutes for departure and 1,584 minutes for arrival. These values should be investigated further during the data cleaning and analysis stages before deciding whether any treatment is required.

## Initial Data Quality Findings

The initial inspection of the raw dataset identified the following points:

- The dataset contains 103,904 rows and 25 columns.
- The `Arrival Delay in Minutes` column contains 310 missing values.
- No completely duplicated rows were found.
- `Unnamed: 0` appears to be an additional index column and may not be required for analysis.
- `id` is a passenger identifier and may not provide useful information for predicting satisfaction.
- The `satisfaction` target contains two categories: `satisfied` and `neutral or dissatisfied`.
- The target distribution is approximately 43.33% satisfied and 56.67% neutral or dissatisfied.
- Service rating variables generally use values between 0 and 5.
- Departure and arrival delays contain some very high values that should be investigated further before deciding whether any treatment is required.

These findings will be considered during the data cleaning and analysis stages.

---

## Conclusions and Next Steps

The raw airline passenger satisfaction dataset was successfully loaded and inspected.

The initial inspection identified 310 missing values in the `Arrival Delay in Minute` column and no completely duplicated records. The `Unnamed: 0` and `id` columns also appear to be identifiers rather than useful analytical features.

The target variable, `satisfaction`, contains two categories and is reasonably distributed between satisfied and neutral or dissatisfied passengers.

The next step is to clean the dataset by addressing the identified data quality issues and preparing the data for further analysis.